src/Hosting/Hosting/src/Internal/HostingApplicationDiagnostics.cs

```diff
@@ -54,8 +56,20 @@ public void BeginRequest(HttpContext httpContext, HostingApplication.Context con
 
             if (loggingEnabled || diagnosticListenerActivityCreationEnabled || _activitySource.HasListeners())
             {
-                context.Activity = StartActivity(httpContext, loggingEnabled, diagnosticListenerActivityCreationEnabled, out var hasDiagnosticListener);
+                context.Activity = StartActivity(httpContext, loggingEnabled, diagnosticListenerActivityCreationEnabled, out var hasDiagnosticListener)!;
                 context.HasDiagnosticListener = hasDiagnosticListener;
+
+                if (context.Activity is not null)
```

Comment:
```suggestion
                if (context.Activity is Activity activity)
```

This does a null check and assigns the property to activity so you can use it further down